packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -447,10 +447,28 @@ export function insertInContainerBefore(
   }
 }
 
+function handleSimulateChildBlur(
+  child: Instance | TextInstance | SuspenseInstance,
+): void {
+  if (
+    enableFlareAPI &&
+    selectionInformation &&
+    child === selectionInformation.focusedElem
+  ) {
+    const simulatedEvent: any = document.createEvent('CustomEvent');
+    simulatedEvent.initCustomEvent('blur', false, true);
+    simulatedEvent.relatedTarget = null;
+    ReactBrowserEventEmitterSetEnabled(true);
+    child.dispatchEvent(simulatedEvent);
```

Comment:
I'm not sure we should actually dispatch on the real DOM. The issue with that is that now other non-React event listeners pick these up. In Firefox/Safari they'd unexpectedly see events that weren't there before. In Chrome they'd start seeing *double*  events. That's the reason I didn't use native dispatching in the replaying system to keep it isolated to React.

I'm also not sure it's the best to dispatch the events in the middle of the mutation sequence since the tree is inconsistent at this point. I'm not 100% sure the event system is resilient to working correctly at this point since we haven't updated to the new tree yet.

The other downside of doing this in here, as opposed to in the restore phase, is that now we add some overhead to all removeChild calls. This used to be inlined but will not need to be outlined and run extra checks for every removed node always. If we do it in the restore phase, it's once per commit instead of once per removed node.

I get that you want this to pick up the path before the deletion happens. However, by definition this node is a React managed node, so you can get the Fiber from it even after it's deleted from the DOM and that gives us the dispatch 
path.

Reply:
So I originally did exactly what you described above: instead of dispatching a native event, I trigger the Flare event responder system with `blur`.

The issue is, by that point the target is no longer in the tree. We need the target to be in the tree to ensure we can properly propagate this event correctly where multiple responders can react to such event – unfortunately by the restore phase fiber tree of the detached element is already deatched too (null return field). This ultimately left me with moving back to this shitty approach.

Reply:
The alternative is that we make this another event other than `blur`, like `reactblur` or something to avoid breaking existing logic.

Reply:
I have another approach...

Reply:
I've addressed in a follow up commit.